content/blog/2019-11-mio-v0.7-alpha.md

```diff
@@ -0,0 +1,201 @@
+access was completely removed in version 0.7 and the iteration API was changed
+to return a reference to `Event` rather then making a copy, but since all
+methods on `Event` only need a reference this shouldn't be a problem.
+
+
+### Removal of the user space queue & deprecated types
+
+Mio version 0.6 had a user space queue, available through the `SetReadiness` and
+`Registration` types, but this has been removed in version 0.7 as it was deemed
+to be outside of the scope for Mio. Users who need a user space queue can use
+one of the queue types found in the [crossbeam] crate.
+
+One use case the user space queue support was waking a thread polling (i.e.
+calling `Poll::poll`) from another thread. To support this use case a new
+`Waker` type was introduced to do exactly this job, calling `Waker::wake` will
+wake the `Poll` it is associated with.
+
+[crossbeam]: https://crates.io/crates/crossbeam
+
+
+### Registering & I/O resource usages changes
+
+By far the biggest user facing change of version 0.7 is the way Mio registers
+I/O sources. In version 0.6 resource would be registered using a `Token`,
+`Ready` interests and `PollOpt`ions. For example the following code would register
+`socket` with readable interests and edge triggers.
+
+```rust
+poll.register(&socket, Token(0), Ready::readable(), PollOpt::edge())?;
+```
+
+As mentioned above the `Event` type was changed to be a wrapper around the
+native OS type, in turn this removed the `Ready` type in favour of having
+methods on `Event` to check for the readiness indicators and getting the
+`Token`. The `Ready` type, as used in registering, was changed to `Interests` to
+better reflect its usage. The API of `Interests` was also changed to take
+advanced of the (somewhat) new associated constants, ensuring that is no longer
+possible to register an event source with empty interests.
+
+In version 0.7 Mio registers all source with edge triggers, removing the need
+for `PollOpt`. How to use edge triggers is explained below.
+
+The trait that defined how to register an event source in version 0.6 was called
+`Evented`, this was changed to `Source` and is now referenced to as
+`event::Source`, as the type lives inside the `event` module. The same three
+methods live on the trait but are changed as described above.
+
+Finally the register functions have moved from `Poll` to a new `Registry` type
+which can be `try_clone`d and used to register sources from different threads.
+All together that means registering the same `socket` with readable interests
+now looks like this:
+
+```rust
+poll.registry().register(&socket, Token(0), Interests::READABLE)?;
+```
+
+
+#### Moving to edge triggers
+
+As already mentioned Mio now registers all I/O source with edge triggers which
+means that users of one-shot and level trigger need to change how they respond
+to events.
+
+The problem was that certain usages of level and one-shot triggers showed
```

Comment:
"The problem" implies an antecedent, but none is apparent.  You seem to be introducing a new problem in this paragraph, so you should say "A problem" instead.

Also, the problem didn't go away with mio 0.7; we just sidestep the issue.  So I would describe the problem in the present tense.

A problem is that certain uses of level and one-shot triggers show differences between OSes that Mio can't cancel without unacceptable overhead.  We want Mio to present a nice cross-platform experience, so we decided to make all I/O sources edge-triggered.  This way the behavior is identical on all platforms.